NadekoBot.Core/Modules/Searches/SearchesSlashCommands.cs

```diff
@@ -0,0 +1,119 @@
+            }
+            
+            var embed = new EmbedBuilder();
+            var data = await _service.GetWeatherDataAsync(query).ConfigureAwait(false);
+            if (data == null)
+            {
+                embed.WithDescription(GetText("city_not_found"))
+                    .WithErrorColor();
+            }
+            else
+            {
+                Func<double, double> f = StandardConversions.CelsiusToFahrenheit;
+                
+                var tz = Context.Guild is null
+                    ? TimeZoneInfo.Utc
+                    : _tzSvc.GetTimeZoneOrUtc(Context.Guild.Id);
+                var sunrise = data.Sys.Sunrise.ToUnixTimestamp();
+                var sunset = data.Sys.Sunset.ToUnixTimestamp();
+                sunrise = sunrise.ToOffset(tz.GetUtcOffset(sunrise));
+                sunset = sunset.ToOffset(tz.GetUtcOffset(sunset));
+                var timezone = $"UTC{sunrise:zzz}";
+
+                embed.AddField(fb => fb.WithName("🌍 " + Format.Bold(GetText("location"))).WithValue($"[{data.Name + ", " + data.Sys.Country}](https://openweathermap.org/city/{data.Id})").WithIsInline(true))
+                    .AddField(fb => fb.WithName("📏 " + Format.Bold(GetText("latlong"))).WithValue($"{data.Coord.Lat}, {data.Coord.Lon}").WithIsInline(true))
+                    .AddField(fb => fb.WithName("☁ " + Format.Bold(GetText("condition"))).WithValue(string.Join(", ", data.Weather.Select(w => w.Main))).WithIsInline(true))
+                    .AddField(fb => fb.WithName("😓 " + Format.Bold(GetText("humidity"))).WithValue($"{data.Main.Humidity}%").WithIsInline(true))
+                    .AddField(fb => fb.WithName("💨 " + Format.Bold(GetText("wind_speed"))).WithValue(data.Wind.Speed + " m/s").WithIsInline(true))
+                    .AddField(fb => fb.WithName("🌡 " + Format.Bold(GetText("temperature"))).WithValue($"{data.Main.Temp:F1}°C / {f(data.Main.Temp):F1}°F").WithIsInline(true))
+                    .AddField(fb => fb.WithName("🔆 " + Format.Bold(GetText("min_max"))).WithValue($"{data.Main.TempMin:F1}°C - {data.Main.TempMax:F1}°C\n{f(data.Main.TempMin):F1}°F - {f(data.Main.TempMax):F1}°F").WithIsInline(true))
+                    .AddField(fb => fb.WithName("🌄 " + Format.Bold(GetText("sunrise"))).WithValue($"{sunrise:HH:mm} {timezone}").WithIsInline(true))
+                    .AddField(fb => fb.WithName("🌇 " + Format.Bold(GetText("sunset"))).WithValue($"{sunset:HH:mm} {timezone}").WithIsInline(true))
+                    .WithOkColor()
+                    .WithFooter(efb => efb.WithText("Powered by openweathermap.org").WithIconUrl($"http://openweathermap.org/img/w/{data.Weather[0].Icon}.png"));
+            }
+            await ctx.Interaction.EmbedAsync(embed).ConfigureAwait(false);
+        }
+
+        public async Task<bool> ValidateQuery(IMessageChannel ch, string query)
+        {
+            if (!string.IsNullOrWhiteSpace(query))
+            {
+                return true;
+            }
+
```

Comment:
Once you move the country codes to a data file/hardcoded in the source, you can easily validate it here to see if there's a validate it here to see if it's a valid country and reject it early. It's better feedback to reject a command with an invalid country than it is to try and return something that will probably not be useful anyway.

Reply:
I added a validate country function here